### DISCUSSION TITLE
Accessibility: Notification-restricted spaces

### DISCUSSION THREAD

**Discussion by joepie91** [element-hq/element-meta#706] [Ideas]
#### What would you like to do?

Configuring a Space such that while it is the active Space, *all* other notifications (both audible and visual representations) outside of that Space are suppressed in Element.

This includes (but may not be limited to):
- Notification popups
- Notification sounds
- Unread notification badge in Element Desktop tray icon
- Notification badge on Space icons
- Invites

#### Why would you like to do it?

With Matrix trying to cover *both* the IM and public chat usecases, it can become quite overwhelming; sometimes, when I [have few 'spoons' left](https://neurodiverging.com/spoons-and-forks-autism-and-fatigue/), I just want to talk to some close and undemanding friends privately, and having notifications/mentions constantly pop up visually for (large) public rooms and DM invites gets very stressful at those moments, as I will feel permanently pressured to respond to them.

Right now, my only 'solution' to that is to close Element *entirely* and not talk to *anybody*. Because I am no longer using a separate chat app for IM and group chat, I no longer have the possibility of talking to friends without *also* being bombarded with notifications elsewhere (which I would normally want to follow up on, but not at that moment). This is obviously not good for my mental health.

#### How would you like to achieve it?

By having a configuration setting for a Space, such as a private one that I have created myself and that only contains non-exhausting people/groups, that makes it "notification-restricted" - whenever it is selected, not only will other rooms and people be hidden, *so will any kind of notifications from them*. Deactivating it (switching to another Space or Home) would un-suppress notifications again.

I believe this can be implemented entirely client-side, so it shouldn't require any protocol changes, and the only thing that needs to be persisted is whether a Space is notification-restricted or not. The default setting does not need to be changed, this would likely only ever be applied by someone to one or two Spaces.

I suspect this would be a comparatively easy change to make, as it mostly just consists of an additional (client-side) notification filter, and (at most) a single persisted setting per Space.

### Have you considered any alternatives?

- __Closing Element:__ This means social isolation, and so is not a good option.
- __Leaving other rooms or changing their notification settings:__ generally, my overwhelmedness is short-lived. It would be excessively impractical to have to change around my rooms or room settings every time I just needed a break from things.
- __Multiple accounts for different purposes:__ Aside from being impractical to have to explain to other people I talk to, Element does not support this anyway.

**Comment by pingiun** [comment]
I think this would be very nice to have, it solves a bunch of notification wants at once

**Comment by kittykat** [comment]
Moving to discussions as this is cross-platform :+1:
